File: examples/catFactFetcher/output.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
)

type FactResponse struct {
	Data []Fact `json:"data"`
}

type Fact struct {
	Fact string `json:"fact"`
}

func GetRandomFact() (string, error) {
	// Send GET request
	resp, err := http.Get("https://meowfacts.herokuapp.com/?count=1")
	if err != nil {
		fmt.Println("Error: Failed to send GET request")
		return "", err
	}
	defer resp.Body.Close()

	// Make sure the http response is 200
	if resp.StatusCode != http.StatusOK {
		fmt.Println("Error: Non-200 HTTP status:", resp.StatusCode)
		return "", fmt.Errorf("Non-200 HTTP status: %v", resp.StatusCode)
	}

	// Decode JSON response
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Error: Failed to read body")
		return "", err
	}

	var factResponse FactResponse
	err = json.Unmarshal(body, &factResponse)
	if err != nil {
		fmt.Println("Error: Failed to decode JSON")
		return "", err
	}

	// Read Fact
	if len(factResponse.Data) == 0 {
		fmt.Println("Error: No facts found in response")
		return "", fmt.Errorf("No facts found in response")
	}
	fact := factResponse.Data[0].Fact

	// Return Fact
	return fact, nil
}

func main() {
	fact, err := GetRandomFact()
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}

	fmt.Println("Random Cat Fact:", fact)
}

File: examples/catFactFetcher/output.js
// Importing the required 'https' module
const https = require('https');

function getRandomFact() {
  // Sending GET request to the API
  https.get('https://meowfacts.herokuapp.com/?count=1', (res) => {
    let data = '';

    // A chunk of data has been received.
    res.on('data', (chunk) => {
      data += chunk;
    });

    // The whole response has been received.
    res.on('end', () => {
      if(res.statusCode === 200) { // Check if the HTTP response is 200
        try {
          // Decode JSON response
          let jsonData = JSON.parse(data);
          
          // Read Fact
          let fact = jsonData.data[0];

          // Return Fact
          return fact;
        } catch (error) {
          console.error('Error:', error); // Log the error and stop the program
        }
      } else {
        console.error('HTTP Response not 200'); // Log the error and stop the program
      }
    });

  }).on('error', (error) => {
    console.error('Error:', error); // Log the error and stop the program
  });
}
